src/pyuvdata/parameter.py

```diff
@@ -917,18 +927,35 @@ def set_from_form(
                 temp_arr = self.value[tuple(val_slice)]
 
                 # Make sure this is a view that is connected in memory
-                assert temp_arr.base is (
-                    self.value if self.value.base is None else self.value.base
-                ), (
-                    "Something is wrong, slicing self.value does not return a view "
-                    "on the original array. Please file an issue in our GitHub "
-                    "issue log so that we can fix it."
-                )
+                if issubclass(self.value.__class__, ShapedLikeNDArray):
+                    for attr in self.value._extra_frameattr_names:
```

Comment:
This actually probably only works for SkyCoord objects, not Time or Quantity objects. Need to add testing for those.

Reply:
I've added testing and I can't figure out how to make this view testing work for Time and Quantity objects. @kartographer any ideas? I'm also wondering whether `np.put` or `np.put_along_axis` might help here?

Reply:
Okay, finally had a chance to poke around with this - yes, it does appear as though for `Quantity` and `Time`, slices and whatnot do not return a view of the original object but rather a fresh new copy.  I don't think this is a situation where `put`/`put_along_axis` is going to help, as the thing that's at issue here is that you've got arrays that are NDArray-like in some ways but not others. ***BUT*** I am able to force the behavior I want if I grab `obj.value` first, which grabs the underlying NDArray and behaves more like as I expect.